src/Meta/class-date-builder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Date Page Metadata Builder class
+ *
+ * @package Parsely
+ * @since 3.4.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely\Meta;
+
+/**
+ * Implements abstract Metadata Builder class to generate the metadata array
+ * for a date page.
+ *
+ * @since 3.4.0
+ */
+class Date_Builder extends Metadata_Builder {
+	/**
+	 * Generates the metadata object by calling the build_* methods and
+	 * returns the value.
+	 *
+	 * @since 3.4.0
+	 *
+	 * @return array<string, mixed>
+	 */
+	public function get_metadata(): array {
+		$this->build_basic();
+		$this->build_headline();
+		$this->build_url();
+
+		return $this->metadata;
+	}
+
+	/**
+	 * Populates the `headline` field in the metadata object.
+	 *
+	 * @since 3.4.0
+	 */
+	private function build_headline(): void {
+		if ( is_year() ) {
+			/* translators: %s: Archive year */
+			$this->metadata['headline'] = sprintf( __( 'Yearly Archive - %s', 'wp-parsely' ), get_the_time( 'Y' ) );
+		} elseif ( is_month() ) {
+			/* translators: %s: Archive month, formatted as F, Y */
+			$this->metadata['headline'] = sprintf( __( 'Monthly Archive - %s', 'wp-parsely' ), get_the_time( 'F, Y' ) );
+		} elseif ( is_day() ) {
+			/* translators: %s: Archive day, formatted as F jS, Y */
+			$this->metadata['headline'] = sprintf( __( 'Daily Archive - %s', 'wp-parsely' ), get_the_time( 'F jS, Y' ) );
```

Comment:
This is a very US-centric date format - so we should consider how we allow these formats to be amended? Perhaps we can also tap into the General Settings site date format?

Reply:
I agree and it's a good point. I'd say that this would change the metadata output, which is not what we want with this PR. 

I've opened an issue to track this and change it separately: https://github.com/Parsely/wp-parsely/issues/850

Reply:
Resolving for this PR.